### ISSUE TITLE
`?` should match a single character

### ISSUE THREAD

**Issue by bytesnz** [kaelzhang/node-ignore#33]
`?` in a pattern should match a single character. Eg. `.*.sw?` should ignore `.testfile.swo`. It's missing from the documentation page you link to for the spec (https://git-scm.com/docs/gitignore), but it is in https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository#Ignoring-Files, and it is the functionality that git (git status) follows.

```
11:05:09 j ~/Development/ls-ignore [master] $ git status
On branch master
Your branch is up-to-date with 'origin/master'.

nothing to commit, working tree clean
11:05:11 j ~/Development/ls-ignore [master] $ cat .gitignore
*~
node_modules
yarn*
11:05:21 j ~/Development/ls-ignore [master] $ touch test.swo
11:05:31 j ~/Development/ls-ignore [master]?1 $ git status
On branch master
Your branch is up-to-date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        test.swo

nothing added to commit but untracked files present (use "git add" to track)
11:05:35 j ~/Development/ls-ignore [master]?1 $ echo "*.sw?" >> .gitignore
11:05:40 j ~/Development/ls-ignore [master]≠1 $ cat .gitignore
*~
node_modules
yarn*
*.sw?
11:05:51 j ~/Development/ls-ignore [master]≠1 $ git status
On branch master
Your branch is up-to-date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   .gitignore

no changes added to commit (use "git add" and/or "git commit -a")
```

```javascript
> nore = ignore().add(['.*.sw?'])
IgnoreBase {
  _rules: 
   [ { origin: '.*.sw?',
       pattern: '.*.sw?',
       negative: false,
       regex: /(?:^|\/)\.[^\/]*\.sw\?(?=$|\/)/i } ],
  _cache: {},
  _added: true,
  [Symbol(node-ignore)]: true }
> nore.filter(['.this.swo'])
[ '.this.swo' ]
> nore.ignores(['.this.swo'])
false
```

**Comment by kaelzhang** [comment]
merged into #35

**Comment by kaelzhang** [comment]
Landed in `4.0.0`
